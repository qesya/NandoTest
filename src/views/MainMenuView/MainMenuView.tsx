import React from 'react';
import {StyleSheet, View, Text, StatusBar} from 'react-native';
import Button from '../../components/Button/Button';
import * as colours from '../../utilities/colours';
import * as fonts from '../../utilities/fonts';

type Props = {
  onPressCustomBridge: () => void;
  onPressRestaurantList: () => void;
};

const MainMenuView = ({onPressCustomBridge, onPressRestaurantList}: Props) => {
  return (
    <View style={styles.container}>
      <StatusBar
        translucent
        backgroundColor={colours.WHITE}
        barStyle="dark-content"
      />
      <View>
        <Text style={styles.txtHeader}>Nando's coding challenge</Text>
        <Button title="Custom Bridge Screen" onPress={onPressCustomBridge} />
        <Button
          title="Restaurant List Screen"
          onPress={onPressRestaurantList}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colours.WHITE,
    paddingHorizontal: 16,
    justifyContent: 'center',
  },
  box: {
    width: 60,
    height: 60,
    marginVertical: 20,
  },
  button: {
    alignSelf: 'center',
    height: 40,
    backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    marginVertical: 10,
  },
  txtHeader: {
    ...fonts.PoppinsLight,
    fontSize: 24,
    textAlign: 'center',
    paddingHorizontal: 20,
    marginBottom: 36,
    color: colours.BLACK,
    textTransform: 'uppercase',
    letterSpacing: 2,
  },
});
export default MainMenuView;
