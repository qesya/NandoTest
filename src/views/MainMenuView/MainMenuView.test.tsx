import React from 'react';
import renderer from 'react-test-renderer';
import MainMenuView from './MainMenuView';

describe('MainMenuView', () => {
  test('renders correctly', () => {
    const component = renderer
      .create(
        <MainMenuView
          onPressCustomBridge={() => jest.fn()}
          onPressRestaurantList={() => jest.fn()}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
