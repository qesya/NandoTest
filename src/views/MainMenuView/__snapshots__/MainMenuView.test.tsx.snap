// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuView renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#1F1F1F",
          "fontFamily": "Poppins-Light",
          "fontSize": 24,
          "letterSpacing": 2,
          "marginBottom": 36,
          "paddingHorizontal": 20,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Nando's coding challenge
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#1F1F1F",
            "borderRadius": 40,
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 10,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
          }
        }
      >
        Custom Bridge Screen
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#1F1F1F",
            "borderRadius": 40,
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 10,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
          }
        }
      >
        Restaurant List Screen
      </Text>
    </View>
  </View>
</View>
`;
